Android/toonDra/app/src/main/java/edu/kaist/mskers/toondra/AutoscrollActivity.java

```diff
@@ -56,6 +71,75 @@ protected void onCreate(Bundle savedInstanceState) {
             "\n" + "Ad mea libris blandit adversarium, ignota inermis instructior ei mei, est no causae alienum verterem. An ius dico nobis feugait, scaevola recteque consequat eu cum. Pro ei sale appetere facilisis. Detracto indoctum te quo, discere dolorum impedit no sed. Sea aperiri honestatis ad, eos tale prompta dolorum no.";
     textarea.setText(temp + temp);
   }
+  private View.OnTouchListener touchEvent = new View.OnTouchListener() {
+    @Override
+    public boolean onTouch(View view, MotionEvent motionEvent) {
+      switch (motionEvent.getAction()) {
+        case MotionEvent.ACTION_DOWN: {
+          countDownTimer.cancel();
+          scrollvelocity = 0;
+          return true;
+        }
+        default: return false;
+      }
+    }
+  };
+  public void onClick(View v) {
+    View view = (View) scrollView.getChildAt(scrollView.getChildCount() - 1);
+    Log.i(TAG, String.valueOf(scrollvelocity));
+    if (view.getBottom() == (scrollView.getHeight() + scrollView.getScrollY())) {
+      scrollvelocity = 0;
+    }
+    if (view.getTop() == scrollView.getScrollY()) {
+      scrollvelocity = 0;
+    }
+    if (countDownTimer != null) countDownTimer.cancel();
+    switch (v.getId()) {
+      case R.id.button_up:
+        scrollvelocity += 3;
+        break;
+      case R.id.button_down:
+        scrollvelocity -= 3;
+        break;
+    }
+    countDownTimer = new CountDownTimer(10000/Math.abs(scrollvelocity), 20) {
```

Comment:
I get ```java.lang.ArithmeticException: divide by zero``` from this line.
This will happen when scrollvelocity == 0.
I suggest a defense case here.